### PR TITLE
Automate uploading of package_esp8266_index.json, add release process doc

### PR DESCRIPTION
**Pull request by igrr**
- http://arduino.esp8266.com/stable/package_esp8266_index.json will now be updated in Travis CI automatically

- Document describing release process (for maintainers) is added

### CODE REVIEW

File: package/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+   ```
+   
+   Push the tag to esp8266/Arduino Github repository.
+
+
+3. Wait for Travis CI build for the tag to pass. Check that the new (draft) release has been created. Check that the boards manager package .zip file has been successfully uploaded as a release artifact.
+
+4. Check that the package index downloaded from http://arduino.esp8266.com/stable/package_esp8266_index.json contains an entry for the new version (it may not be the first one).
+
+5. Edit release description, paste release notes. Remove "draft" status of the release and publish it.
```
Comment (devyte) [inline package/README.md]: I assume this means in github, in the "Releases" tab -> Edit button?

**Comment by earlephilhower** [comment]
I won't be able to review this deeply until tonight, but doesn't the including of a SSH key in the scripts mean that anyone can now upload to the server just by copying the key?  I'm thinking of recent hacks where miscreants were able to put up malicious packages in real repos...

File: package/README.md (hunk 1 above)
Comment (igrr) [reply package/README.md]: Yes, i'll update the wording

**Comment by igrr** [comment]
@earlephilhower Included in the repo are:

* encrypted private key, for the SSH client. Encryption key is stored in Travis CI variables and is not accessible outside (see https://docs.travis-ci.com/user/encrypting-files/). This is decrypted when it's time to deploy.

* public key of the SSH server. This is to prevent MITM where SSH client would be connecting to someone who is pretending to be arduino.esp8266.com. Since this is the public key, no harm in revealing it. This is similar to including server public key (or fingerprint) into an HTTPS sketch.

**Comment by earlephilhower** [comment]
Great, @igrr.  Thanks for the info.  I'll read it thru tonight, but if @devyte thinks its OK then we're probably good to go anyway.

File: package/README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+   *  Pack source files into a zip file, excluding some files and tweaking platform.txt.
+   *  Download current latest package index json file from Github Releases. This file contains descriptions of all previous releases.
+   *  Generate package index for the new release.
+   *  Combines new release with previous releases in one json file (using `merge_packages.py` script).
+
+4. Travis CI uploads boards manager package (.zip file) and package index (.json file) to Github Releases, creating a draft release at the same time.
+
+5. Travis CI also uploads package index .json file to `http://arduino.esp8266.com/stable/package_esp8266_index.json`, i.e. well-known URL used by most users. 
+
+6. When the draft release is created, maintainer edits release description and inserts changelog into the description field, unmarks the release as draft, and publishes the release.
```
Comment (earlephilhower) [inline package/README.md]: I think I understand everything except this.  How/where is this draft release created?

**Pull request merged**